kafka/adminapi.go

```diff
@@ -2197,8 +2197,12 @@ func (a *AdminClient) ListConsumerGroupOffsets(
 	for i, groupPartitions := range groupsPartitions {
 		// We need to destroy this list because rd_kafka_ListConsumerGroupOffsets_new
 		// creates a copy of it.
-		cPartitions := newCPartsFromTopicPartitions(groupPartitions.Partitions)
-		defer C.rd_kafka_topic_partition_list_destroy(cPartitions)
+		var cPartitions *C.rd_kafka_topic_partition_list_t = nil
+
+		if len(groupPartitions.Partitions) > 0 {
```

Comment:
Change this check to 
`groupPartitions.Partitions != nil`.
This is because the underlying library differentiates between a 0 sized list and a nil topic partition list. The former is an error, but the latter stands for all the partitions.

Reply:
@milindl thanks for your review. I'll change this and add some unit tests for this case